src/Uno.UI/UI/Xaml/Controls/DatePicker/DatePicker.cs

```diff
@@ -237,56 +270,69 @@ private void SetupFlyoutButton()
 			if (_flyoutButton != null)
 			{
 #if __IOS__ || __ANDROID__
-				_flyoutButton.Flyout = new DatePickerFlyout()
+				_flyoutButton.Flyout = _flyout = new DatePickerFlyout()
 				{
-#if __IOS__
-					Placement = FlyoutPlacement,
-#endif
 					Date = Date,
 					MinYear = MinYear,
-					MaxYear = MaxYear
+					MaxYear = MaxYear,
 				};
 
-				BindToFlyout(nameof(Date));
+				_flyout.Opened += (s, e) => _flyout.Date = SelectedDate ?? DateTimeOffset.Now;
+				_flyout.DatePicked += (s, e) => Date = e.NewDate;
+
 				BindToFlyout(nameof(MinYear));
 				BindToFlyout(nameof(MaxYear));
-				_flyoutButton.Flyout.BindToEquivalentProperty(this, nameof(LightDismissOverlayMode));
-				_flyoutButton.Flyout.BindToEquivalentProperty(this, nameof(LightDismissOverlayBackground));
+				_flyout.BindToEquivalentProperty(this, nameof(LightDismissOverlayMode));
+				_flyout.BindToEquivalentProperty(this, nameof(LightDismissOverlayBackground));
 #endif
 			}
 		}
 
 		private void BindToFlyout(string propertyName)
 		{
-			this.Binding(propertyName, propertyName, _flyoutButton.Flyout, BindingMode.TwoWay);
+			this.Binding(propertyName, propertyName, _flyout, BindingMode.TwoWay);
 		}
 
 		private void InitializeTextBlocks(IFrameworkElement container)
 		{
-			var items = GetOrderedTextBlocksForCulture(CultureInfo.CurrentCulture);
-			var flyoutButtonGrid = container.GetTemplateChild(FlyoutButtonGridPartName) as Grid;
-
-			Grid.SetColumn(items[0], GetColumnIndex(flyoutButtonGrid, DayColumnPartName));
-			Grid.SetColumn(items[1], GetColumnIndex(flyoutButtonGrid, MonthColumnPartName));
-			Grid.SetColumn(items[2], GetColumnIndex(flyoutButtonGrid, YearColumnPartName));
+			if (container.GetTemplateChild(FlyoutButtonGridPartName) is Grid grid)
+			{
+				var items = GetOrderedTextBlocksForCulture(CultureInfo.CurrentCulture);
+				var oldColumnInfos = new[] { DayColumnPartName, MonthColumnPartName, YearColumnPartName }
```

Comment:
I have no idea what the modified code here is meant to be doing, can you add some comments?